src/main/java/net/dv8tion/jda/internal/components/mediagallery/MediaGalleryItemImpl.java

```diff
@@ -0,0 +1,150 @@
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.components.mediagallery;
+
+import net.dv8tion.jda.api.components.ResolvedMedia;
+import net.dv8tion.jda.api.components.mediagallery.MediaGalleryItem;
+import net.dv8tion.jda.api.utils.FileUpload;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.components.ResolvedMediaImpl;
+import net.dv8tion.jda.internal.entities.FileContainerMixin;
+import net.dv8tion.jda.internal.utils.Checks;
+import net.dv8tion.jda.internal.utils.EntityString;
+import net.dv8tion.jda.internal.utils.Helpers;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+public class MediaGalleryItemImpl implements MediaGalleryItem, FileContainerMixin
+{
+    private final String url, description;
+    private final ResolvedMedia media;
+    private final boolean spoiler;
+
+    public MediaGalleryItemImpl(DataObject obj)
+    {
+        this(
+                obj.getObject("media").getString("url"),
+                obj.getString("description", null),
+                new ResolvedMediaImpl(obj.getObject("media")),
+                obj.getBoolean("spoiler", false)
+        );
+    }
+
+    public MediaGalleryItemImpl(String url)
+    {
+        this(url, null, null, false);
+    }
+
+    public MediaGalleryItemImpl(String url, String description, ResolvedMedia media, boolean spoiler)
+    {
+        this.url = url;
+        this.media = media;
+        this.description = description;
+        this.spoiler = spoiler;
+    }
+
+    @Nonnull
+    @Override
+    public MediaGalleryItem withDescription(@Nullable String description)
+    {
+        if (description != null)
+        {
+            Checks.notBlank(description, "Description");
+            Checks.notLonger(description, MAX_DESCRIPTION_LENGTH, "Description");
+        }
+        return new MediaGalleryItemImpl(url, description, media, spoiler);
+    }
+
+    @Nonnull
+    @Override
+    public MediaGalleryItem withSpoiler(boolean spoiler)
+    {
+        return new MediaGalleryItemImpl(url, description, media, spoiler);
+    }
+
+    @Nonnull
+    @Override
+    public String getUrl()
+    {
+        return url;
+    }
+
+    @Nullable
+    @Override
+    public ResolvedMedia getResolvedMedia()
+    {
+        return media;
+    }
+
+    @Override
+    public Stream<FileUpload> getFiles() {
+        final String fileName = Helpers.getLastPathSegment(media.getUrl());
+        return Stream.of(media.getProxy().downloadAsFileUpload(fileName));
+    }
+
+    @Nullable
+    @Override
+    public String getDescription()
+    {
+        return description;
+    }
+
+    @Override
+    public boolean isSpoiler()
+    {
+        return spoiler;
+    }
+
+    @Nonnull
+    @Override
+    public DataObject toData()
+    {
+        final String fileName = Helpers.getLastPathSegment(media.getUrl());
```

Comment:
This causes a NullPointerException when creating a message that has a media gallery item

`jda-ktx` calls `MediaGalleryItem.fromUrl(url)`, which that calls the `MediaGalleryItemImpl(String url)` constructor, and the `media` parameter is set to null

```java
        final String mediaUrl;
        if (this.media != null) {
            final String fileName = Helpers.getLastPathSegment(media.getUrl());
            mediaUrl = "attachment://" + fileName;
        } else {
            mediaUrl = this.url;
        }
```